site/content/blog/2025-02-17-portswigger-lab-walkthrough-broken-brute-force-protection-ip-block/index.md

```diff
@@ -0,0 +1,793 @@
+
+You can potentially bypass rate-limiting and IP blocking in other ways, like making [multiple password guesses in a single request](https://portswigger.net/web-security/authentication/password-based/lab-broken-brute-force-protection-multiple-credentials-per-request). However, we’ll focus on the three methods listed above.
+
+Based on the title of this lab, we’re dealing with an IP block, so we’ll attempt to bypass this. Let’s make a few requests in the Request Editor (ZAP’s equivalent of Burp Suite’s Repeater).
+
+Head over to ZAP and open the POST request we captured earlier. You’ll find it in the Sites Tree under the *Sites* tab. It’ll look something like “POST:login()(password, username)”:
+
+&nbsp;
+![open-with-request-editor.png](images/open-with-request-editor.png)
+&nbsp;
+
+We want to see how many requests we can make in quick succession before our IP address gets blocked. After four login attempts, we get an error message: **“You have made too many incorrect login attempts. Please try again in 1 minute(s).”**
+
+After one minute, we can  indeed make more login attempts. The idea is that if we fail to log in three times in a row, the server activates an IP block. It then denies any more requests sent for the next minute.
+
+If we enter our own credentials, `wiener:peter`, on every third login attempt, the IP block never activates. Each successful login resets the counter tracking the number of failed login attempts. We can keep going until we’ve tried enough passwords from our wordlist to find the right one for Carlos's account.
+
+> [!NOTE]
+> This bypass — taking advantage of a logic flaw to reset the failure counter —  enables us to brute-force Carlos’ password and successfully solve this lab. Although not the point of the lab, we can also wait out the IP block since we’re dealing with only a few credentials. This is covered in “Method 2: Wait Out the IP Block”.
+
+What about changing our IP address? I used the “X-Forwarded-For” header successfully in another lab. But the server here appears to block IPs based on the client's actual IP address (determined by the network layer rather than HTTP headers from the application layer), which makes sense in the context of rate-limiting. Rotating IP addresses via a proxy or VPN could work, but we won’t explore that here.
+
+## Brute-forcing the Password
+
+The default attack style of the ZAP Fuzzer when multiple payload positions are assigned a payload set is the [cluster bomb attack](https://portswigger.net/burp/documentation/desktop/tools/intruder/configure-attack/attack-types). (The ZAP Fuzzer is equivalent to Burp Suite’s Intruder.) A cluster bomb attack iterates through all possible combinations of the payloads. However, we want a [pitchfork attack](https://portswigger.net/burp/documentation/desktop/tools/intruder/configure-attack/attack-types) to run through our username and password lists simultaneously. So, we’re using custom scripts in both of the methods below.
```

Comment:
I thought you said earlier we knew for sure that it was "carlos" and we only needed to sort out the password?

Reply:
Yeah, but then we also need to login to our own account (`wiener:peter`) every third login attempt.